src/object/object_primitive.c

```diff
@@ -16374,3 +16381,225 @@ mr_get_compression_length (const char *string, int charlen)
 
   return length;
 }
+
+
+/*
+ * mr_get_size_and_write_string_to_buffer ()
+ *	  			  : Writes a VARCHAR or VARNCHAR to buffer, without needing the buffer initialized.
+ *				    
+ * buf(out)			  : Buffer to be written to.
+ * val_p(in)			  : Memory area to be written to.
+ * value(in)			  : DB_VALUE to be written.
+ * val_size(out)		  : Disk size of the DB_VALUE.
+ * align(in)			  : 
+ *
+ *  Note:
+ *	We use this to avoid double compression when it is required to have the size of the DB_VALUE, previous
+ *	to the write of the DB_VALUE in the buffer.
+ */
+
+int
+mr_get_size_and_write_string_to_buffer (OR_BUF * buf, char *val_p, DB_VALUE * value, int *val_size, int align)
+{
+  char *compressed_string = NULL, *string = NULL, *str = NULL;
+  int rc = NO_ERROR, str_length = 0, length = 0;
+  lzo_uint compression_length = 0;
+  lzo_bytep wrkmem = NULL;
+  bool compressed = false;
+  int error_abort = 0;
+
+  /* Checks to be sure that we have the correct input */
+  assert (DB_VALUE_DOMAIN_TYPE (value) == DB_TYPE_VARNCHAR || DB_VALUE_DOMAIN_TYPE (value) == DB_TYPE_STRING);
+  assert (DB_GET_STRING_SIZE (value) >= PRIM_MINIMUM_STRING_LENGTH_FOR_COMPRESSION);
+
+  error_abort = buf->error_abort;
```

Comment:
@lucianlazar2211 

Please notice reading `buf->error_abort` would be UMR(Uninitialized Memory Read).
I don't know all the callers of the function does not initialize `buf` yet but `qdata_copy_db_value_to_tuple_value` does not. I guess none does.